src/legacy/state/orders/middleware/composableOrdersPopupMiddleware.ts

```diff
@@ -0,0 +1,39 @@
+import { isAnyOf } from '@reduxjs/toolkit'
+import { jotaiStore } from 'jotaiStore'
+import { Middleware } from 'redux'
+
+import { isTruthy } from 'legacy/utils/misc'
+
+import { tokensByAddressAtom } from 'modules/tokensList/state/tokensListAtom'
+import { setTwapOrderStatusAtom, twapOrdersAtom } from 'modules/twap/state/twapOrdersListAtom'
+import { TwapOrderStatus } from 'modules/twap/types'
+import { mapTwapOrderToStoreOrder } from 'modules/twap/utils/mapTwapOrderToStoreOrder'
+
+import { batchCancelOrdersPopup } from './batchCancelOrdersPopup'
+
+import { AppState } from '../../index'
+import { cancelOrdersBatch } from '../actions'
+
+const isCancelOrdersBatch = isAnyOf(cancelOrdersBatch)
+
+export const composableOrdersPopupMiddleware: Middleware<Record<string, unknown>, AppState> =
+  (store) => (next) => (action) => {
+    const result = next(action)
+
+    if (isCancelOrdersBatch(action)) {
+      // TODO: generalize this to all composable orders, not only twap
+      const composableOrders = jotaiStore.get(twapOrdersAtom)
+      const tokensByAddress = jotaiStore.get(tokensByAddressAtom)
+
+      const cancelledOrdersItems = action.payload.ids.map((id) => composableOrders[id]).filter(isTruthy)
```

Comment:
It would be good to add some comment explaining what is happening here.

Reply:
Added more comments